pkg/agent/flowexporter/connections/connections.go

```diff
@@ -0,0 +1,120 @@
+package connections
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/flowexporter"
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+)
+
+var _ CxnStoreAndBuilder = new(cxnStoreAndBuilder)
+
+type CxnStoreAndBuilder interface {
+	Run(stopCh <-chan struct{})
+}
+
+// TODO: Name dilemma--cxnStoreAndBuilder or cxnCache or something else?
+type cxnStoreAndBuilder struct {
+	cxns            map[[5]string]flowexporter.Connection // Add 5-tuple as string array
+	connTrackPoller ConnTrackPoller
+	ifaceStore      interfacestore.InterfaceStore
+	mutex           sync.Mutex
+}
+
+func NewCxnStoreAndBuilder(connTrack ConnTrackPoller, ifaceStore interfacestore.InterfaceStore) *cxnStoreAndBuilder {
+	return &cxnStoreAndBuilder{
+		cxns:            make(map[[5]string]flowexporter.Connection),
+		connTrackPoller: connTrack,
+		ifaceStore:      ifaceStore,
+	}
+}
+
+func (csb *cxnStoreAndBuilder) Run(stopCh <-chan struct{}) {
+	klog.Infof("Starting conntrack polling")
+	for {
+		select {
+		case <-stopCh:
+			break
+		case <-time.After(flowexporter.PollInterval):
+			_, err := csb.poll()
+			if err != nil {
+				klog.Errorf("Error during conntrack poll cycle: %v", err)
+			}
+		}
+	}
+}
+
+// Update Connection map; If Connection is already present, update timestamp, counters etc.,
+// or add new Connection by 5-tuple of original flow along with local pod and podNameSpace.
+func (csb *cxnStoreAndBuilder) addOrUpdateCxn(cxn *flowexporter.Connection) {
+	flowTuple := [5]string{cxn.TupleOrig.SourceAddress.String(),
+		strconv.FormatUint(uint64(cxn.TupleOrig.SourcePort), 10),
+		cxn.TupleReply.SourceAddress.String(),
+		strconv.FormatUint(uint64(cxn.TupleReply.SourcePort), 10),
+		strconv.FormatUint(uint64(cxn.TupleOrig.Protocol), 10),
+	}
+
+	existingFlow, exists := csb.getCxnByFlowTuple(flowTuple)
+
+	csb.mutex.Lock()
+	defer csb.mutex.Unlock()
+	if exists {
+		// Update the Antrea flow. Update only fields that are used in IPFIX record
+		// Can same 5-tuple flow get deleted and added to conntrack table? If so use ID.
+		existingFlow.StopTime = cxn.StopTime
+		existingFlow.OriginalBytes = cxn.OriginalBytes
+		existingFlow.OriginalPackets = cxn.OriginalPackets
+		existingFlow.ReverseBytes = cxn.ReverseBytes
+		existingFlow.ReversePackets = cxn.ReversePackets
+		// Reassign the flow to update the map
+		csb.cxns[flowTuple] = *existingFlow
+	} else {
+		var srcFound, dstFound bool
+		sIface, srcFound := csb.ifaceStore.GetInterfaceByIP(cxn.TupleOrig.SourceAddress.String())
+		dIface, dstFound := csb.ifaceStore.GetInterfaceByIP(cxn.TupleReply.SourceAddress.String())
+		if !srcFound && !dstFound {
+			klog.Warningf("Cannot map any of the IP %s or %s to a local pod", cxn.TupleOrig.SourceAddress.String(), cxn.TupleReply.SourceAddress.String())
```

Comment:
did you mean to have an early `return` statement here?

Reply:
This was intentional. I did not want connection addition to fail, if podIP cannot be mapped to a name.